core/templates/dev/head/pages/collection_editor/editor_tab/CollectionNodeCreatorDirective.js

```diff
@@ -25,19 +25,47 @@ oppia.directive('collectionNodeCreator', [function() {
       'validatorsService', 'CollectionEditorStateService',
       'CollectionLinearizerService', 'CollectionUpdateService',
       'CollectionNodeObjectFactory', 'ExplorationSummaryBackendApiService',
-      'siteAnalyticsService',
+      'siteAnalyticsService', 'UrlInterpolationService',
       function(
           $scope, $http, $window, $filter, alertsService,
           validatorsService, CollectionEditorStateService,
           CollectionLinearizerService, CollectionUpdateService,
           CollectionNodeObjectFactory, ExplorationSummaryBackendApiService,
-          siteAnalyticsService) {
+          siteAnalyticsService, UrlInterpolationService) {
         $scope.collection = CollectionEditorStateService.getCollection();
         $scope.newExplorationId = '';
         $scope.newExplorationTitle = '';
         var CREATE_NEW_EXPLORATION_URL_TEMPLATE = '/create/<exploration_id>';
+        var SEARCH_EXPLORATION_URL_TEMPLATE = (
+          '/exploration/metadata_search?q=<search_query>');
+
+        $scope.getExplorations = function(searchQuery) {
+          if (/^[a-zA-Z0-9- ]*$/.test(searchQuery) == true) {
```

Comment:
No need to have `== true`, since the `.test()` returns either `True` or `False. Also, it would be nice to either separate this test out into a separate function so that it's easier to digest.